### ISSUE TITLE
Lint/LiteralInInterpolation incorrect autocorrect for escaped string content in interpolated hashes

### ISSUE THREAD

**Issue by RedZapdos123** [rubocop/rubocop#15298]
## Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

## Issue type

Incorrect autocorrect

## Cop name

`Lint/LiteralInInterpolation`

## Steps to reproduce the problem

I reproduced this on the latest stable RuboCop release and also on current `master` at `1f83e39bc2963f7d54fd8eebe6f28c49a94de0c4`.

This looks related to #15250 and #15291, but it still affects string keys and values nested inside interpolated hashes.

### Repro 1: escaped interpolation text in hash value becomes live interpolation

1. Create a file `example.rb` with:

```ruby
puts "#{ { foo: "\#{bar}" } }"
```

2. Run:

```sh
bundle exec ruby exe/rubocop example.rb --only Lint/LiteralInInterpolation -A
ruby example.rb
```

### Repro 2: escaped interpolation text in hash key becomes live interpolation

1. Create a file `example.rb` with:

```ruby
puts "#{ { "\#{bar}" => 1 } }"
```

2. Run:

```sh
bundle exec ruby exe/rubocop example.rb --only Lint/LiteralInInterpolation -A
ruby example.rb
```

### Repro 3: backslashes in nested hash string values are reinterpreted

1. Create a file `example.rb` with:

```ruby
puts "#{ { foo: "\\bar" } }"
```

2. Run:

```sh
bundle exec ruby exe/rubocop example.rb --only Lint/LiteralInInterpolation -A
ruby example.rb
```

### Repro 4: escaped instance variable interpolation text in hash values is reinterpreted

1. Create a file `example.rb` with:

```ruby
puts "#{ { foo: "\#@bar" } }"
```

2. Run:

```sh
bundle exec ruby exe/rubocop example.rb --only Lint/LiteralInInterpolation -A
ruby example.rb
```

## Expected behavior

Autocorrect should preserve the stringified hash representation, or skip autocorrection if the nested string content cannot be serialized safely.

For these cases:

- `"\#{bar}"` should stay escaped inside the resulting stringified hash output
- `"\\bar"` should keep the same backslash content after autocorrect
- `"\#@bar"` should stay escaped inside the resulting stringified hash output

### Repro 1

Before autocorrect, the program prints:

```text
{:foo=>"\#{bar}"}
```

After autocorrect, RuboCop rewrites the file to:

```ruby
puts "{:foo=>\"#{bar}\"}"
```

and running it raises:

```text
undefined local variable or method `bar'
```

### Repro 2

Before autocorrect, the program prints:

```text
{"\#{bar}"=>1}
```

After autocorrect, RuboCop rewrites the file to:

```ruby
puts "{\"#{bar}\"=>1}"
```

and running it raises:

```text
undefined local variable or method `bar'
```

### Repro 3

Before autocorrect, the program prints:

```text
{:foo=>"\\bar"}
```

After autocorrect, RuboCop rewrites the file to:

```ruby
puts "{:foo=>\"\bar\"}"
```

and the runtime output changes because the backslash is reinterpreted.

### Repro 4

Before autocorrect, the program prints:

```text
{:foo=>"\#@bar"}
```

After autocorrect, RuboCop rewrites the file to:

```ruby
puts "{:foo=>\"#@bar\"}"
```

and running it changes the output to:

```text
{:foo=>""}
```

This seems to happen because hash child strings are still serialized with narrower escaping than top-level string literals.

## RuboCop configuration

No special configuration.

## RuboCop version

```shell
$ bundle exec ruby exe/rubocop -V
1.88.0 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 3.2.3) +Rubydex [x86_64-linux-gnu]
  - rubocop-performance 1.26.1
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.10.2
```
